code/game/objects/structures/signs/_signs.dm

```diff
@@ -30,111 +36,133 @@
-			if("Blank")
-				sign_type = /obj/structure/sign/basic
-			if("Secure Area")
-				sign_type = /obj/structure/sign/warning/securearea
-			if("Biohazard")
-				sign_type = /obj/structure/sign/warning/biohazard
-			if("High Voltage")
-				sign_type = /obj/structure/sign/warning/electricshock
-			if("Radiation")
-				sign_type = /obj/structure/sign/warning/radiation
-			if("Hard Vacuum Ahead")
-				sign_type = /obj/structure/sign/warning/vacuum
-			if("Disposal: Leads To Space")
-				sign_type = /obj/structure/sign/warning/deathsposal
-			if("Danger: Fire")
-				sign_type = /obj/structure/sign/warning/fire
-			if("No Smoking")
-				sign_type = /obj/structure/sign/warning/nosmoking/circle
-			if("Test Chamber")
-				sign_type = /obj/structure/sign/warning/testchamber
-			if("Firing Range")
-				sign_type = /obj/structure/sign/warning/firingrange
-			if("Extreme Cold")
-				sign_type = /obj/structure/sign/warning/coldtemp
-			if("Extreme Heat")
-				sign_type = /obj/structure/sign/warning/hottemp
-			if("Gas Mask")
-				sign_type = /obj/structure/sign/warning/gasmask
-			if("Reactive Chemicals")
-				sign_type = /obj/structure/sign/warning/chemdiamond
-			if("Medbay")
-				sign_type = /obj/structure/sign/departments/medbay/alt
-			if("Science")
-				sign_type = /obj/structure/sign/departments/science
-			if("Chemistry")
-				sign_type = /obj/structure/sign/departments/chemistry
-			if("Hydroponics")
-				sign_type = /obj/structure/sign/departments/botany
-			if("Xenobiology")
-				sign_type = /obj/structure/sign/departments/xenobio
-			if("Nanites Lab")
-				sign_type = /obj/structure/sign/departments/nanites
-			if("Maintenance")
-				sign_type = /obj/structure/sign/departments/mait
+		SB.obj_integrity = obj_integrity //Transfer how damaged it is.
+		SB.setDir(dir)
+		qdel(src) //The sign structure on the wall goes poof and only the sign item from unwrenching remains.
+	return TRUE
 
-		//Make sure user is adjacent still
-		if(!Adjacent(user))
+/obj/structure/sign/attackby(obj/item/I, mob/user, params)
+	if(is_editable && istype(I, /obj/item/pen))
+		if(!length(GLOB.editable_sign_types))
+			populate_editable_sign_types()
+			if(!length(GLOB.editable_sign_types))
+				CRASH("GLOB.editable_sign_types failed to populate")
+		var/choice = input(user, "Select a sign type.", "Sign Customization") as null|anything in GLOB.editable_sign_types
+		if(!choice)
 			return
-
-		if(!sign_type)
+		if(!Adjacent(user)) //Make sure user is adjacent still.
+			to_chat(user, "<span class='warning'>You need to stand next to the sign to change it!</span>")
 			return
-
-		//It's import to clone the pixel layout information
+		user.visible_message("<span class='notice'>[user] begins changing [src].</span>", \
+							 "<span class='notice'>You begin changing [src].</span>")
+		if(!do_after(user, 40, target = src)) //Small delay for changing signs instead of it being instant, so somebody could be shoved or stunned to prevent them from doing so.
```

Comment:
```suggestion
		if(!do_after(user, 4 SECONDS, target = src)) //Small delay for changing signs instead of it being instant, so somebody could be shoved or stunned to prevent them from doing so.
```

Reply:
Also done.